### ISSUE TITLE
Segfault when adding when adding upnp as video

### ISSUE THREAD

**Issue by c0dysharma** [xbmc/xbmc#19948]
### Describe the bug
Here is a clear and concise description of what the problem is:

Getting segfault when adding upnp server as video. Server is running on my mobile phone.
I am using kodi-x11- https://archlinux.org/packages/community/x86_64/kodi-x11/

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

It should add the server and show videos

## Actual Behavior

Segmentation fault

### To Reproduce

1.  Create upnp server on other device (preferred android)with kodi app/pkg. 
2. Get the other device's IP address.
3. Add upnp server with that IP as video source

### Debuglog

The debuglog can be found here: 
kodi.old.log- https://termbin.com/wxm2i
kodi.log- https://termbin.com/fxpi
crash  report - https://termbin.com/3jo0

### Screenshots 
Here are some links or screenshots to help explain the problem:

Crashes as soon as I hit OK (see picture) https://imgur.com/a/wewiZY2

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Manjaro Linux
 - Kodi version: kodi-x11 19.1-1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by lrusak** [comment]
The crash is here
```

Thread 1 (Thread 0x7efeec569ac0 (LWP 2311189)):
#0  0x00007efef0ee9515 in __strlen_avx2 () from /usr/lib/libc.so.6
#1  0x00005633dff77126 in CUtil::GetTitleFromPath[abi:cxx11](CURL const&, bool) ()

```

**Comment by c0dysharma** [comment]
> The crash is here
> 
> ```
> 
> Thread 1 (Thread 0x7efeec569ac0 (LWP 2311189)):
> #0  0x00007efef0ee9515 in __strlen_avx2 () from /usr/lib/libc.so.6
> #1  0x00005633dff77126 in CUtil::GetTitleFromPath[abi:cxx11](CURL const&, bool) ()
> ```

Um, Sorry but can you suggest a workaround? I am already on latest version of glibc https://archlinux.org/packages/core/x86_64/glibc/

**Comment by wsnipex** [comment]
does you CPU support avx2?

**Comment by c0dysharma** [comment]
> does you CPU support avx2?

Sorry for late reply and yes it does 
![image](https://user-images.githubusercontent.com/55319022/127727191-2ffc7c77-7b44-4522-b652-e7e7a711eb7f.png)

**Comment by arnova** [comment]
Backtrace makes no sense. Stack is probably already smashed before the dump is generated...

**Comment by neo1973** [comment]
This has been fixed with #24215.
